python/Scripts/creatematerial.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+__doc__ = 'Create materialX file conatin standard surface uber shader with input texture for given directory'
+
+import os
+import re
+import json
+import argparse
+
+import logging
+from typing import List, Dict
+
+import MaterialX
+import udimutils
+
+# Configure the logger
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+
+# Create a logger instance
+logger = logging.getLogger('creatematerial')
+
+
+def get_cfg() -> Dict:
+    """
+    Get the patterns and settings from cfg file
+    return: dict of resultant settings
+    """
+    json_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'textures_patterns.json')
+    # TODO: Add option here to take the user overrides customization file
+    with open(json_path, 'r') as f:
```

Comment:
Just a paranoia check but the script will halt if the file cannot be found.
Would it be better to get this once and if it fails not even start ? Not sure how you feel about a global, but the other option could be to create a class for this, then the methods can reference class data ?

Reply:
Actually, I placed the json file here because I didn't know what the best place of this file. 
yah.. its a good idea to make a function that cooks all configuration and overrides and returns the settings directly. I will work on it, but my plan to make a default file will be released with the script, and the end user will have all rights to override or extend what he wants in separate files.
What are your thoughts about that?